test/unit/custom_index_test.lua

```diff
@@ -204,8 +204,50 @@ function g.test_tree_index_multikey(cg)
     task:kill()
 end
 
+function g.test_memtx_tree_functional_index_broken(cg)
+    t.skip_if(_TARANTOOL < "2" or not helpers.memtx_func_index_is_broken(),
+              "Unsupported Tarantool version")
+    t.skip_if(cg.params.index_type ~= 'TREE', 'Unsupported index type')
+    t.skip_if(cg.params.engine == 'vinyl', 'Unsupported engine')
+    local expected = "Functional indices are not supported for Tarantool < 2.8.4," ..
+                     " see options.force_broken"
+
+    t.assert_error_msg_contains(expected, expirationd.start, "clean_all", cg.space.id,
+            helpers.is_expired_debug, {index = "functional_index"})
+end
+
+function g.test_memtx_tree_functional_index_force_broken(cg)
```

Comment:
I afraid this test may be unstable / crashing. Can we run the case on an empty space (just to check that the error is not raised)? I guess it should be safe.

Reply:
Ah, I see now: you select all tuples at once before all deletions. Good solution.

It would be good to mark this place with a short comment. Not obvious at first glance.

Reply:
Thank you, I rewrote the test: insert + expirationd.start + checks + truncate. Now it's obvious.

Reply:
It seems you understood me wrongly. I'm about the 'force broken' test case: when the `force_allow_functional_index` is enabled. Especially, about this chunk of the code:

```lua
local select_with = function(task)
    return pairs(task.index:select({}, {iterator = "ALL"}))
end
```

I just want a short comment, why it is necessary.

Reply:
Yes, sorry for the misunderstanding. I've added a comment:
```patch
--- a/test/unit/custom_index_test.lua
+++ b/test/unit/custom_index_test.lua
@@ -236,8 +236,11 @@ function g.test_memtx_tree_functional_index_force_broken(cg)
     cg.space:insert({1, "1", nil, nil, nil, nil, nil, "12"})
     cg.space:insert({2, "2", nil, nil, nil, nil, nil, "21"})
 
+    -- The problem occurs when we iterate through a functional index and delete
+    -- a current tuple. A possible solution is somehow to process tuples chunk
+    -- by chunk using select calls instead of iterating with index:pairs().
     local select_with = function(task)
-        return pairs(task.index:select({}, {iterator = "ALL"}))
+        return pairs(task.index:select({}, {iterator = "ALL", limit = 100}))
     end
```

Reply:
Thanks for the update!